core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -570,162 +689,129 @@ private BlockStoreLocation commitBlockInternal(long sessionId, long blockId, boo
-      throws BlockDoesNotExistException {
-    // NOTE: a temp block is supposed to be visible for its own writer, unnecessary to acquire
-    // block lock here since no sharing
-    try (LockResource r = new LockResource(mMetadataWriteLock)) {
-      TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
-      if (tempBlockMeta.getParentDir().getAvailableBytes() < additionalBytes) {
-        return new Pair<>(false, tempBlockMeta.getBlockLocation());
-      }
-      // Increase the size of this temp block
-      try {
-        mMetaManager.resizeTempBlockMeta(tempBlockMeta,
-            tempBlockMeta.getBlockSize() + additionalBytes);
-      } catch (InvalidWorkerStateException e) {
-        throw Throwables.propagate(e); // we shall never reach here
-      }
-      return new Pair<>(true, null);
-    }
-  }
-
-  /**
-   * Tries to get an eviction plan to free a certain amount of space in the given location, and
-   * carries out this plan with the best effort.
+   * Tries to free a certain amount of space in the given location.
    *
    * @param sessionId the session id
-   * @param availableBytes amount of space in bytes to free
+   * @param minContigiousBytes the minimum amount of contigious space in bytes to set available
+   * @param minAvailableBytes the minimum amount of space in bytes to set available
    * @param location location of space
-   * @param mode the eviction mode
-   * @throws WorkerOutOfSpaceException if it is impossible to achieve the free requirement
+   * @throws WorkerOutOfSpaceException if it is impossible to achieve minimum space requirement
    */
-  private void freeSpaceInternal(long sessionId, long availableBytes, BlockStoreLocation location,
-      Evictor.Mode mode) throws WorkerOutOfSpaceException, IOException {
-    EvictionPlan plan;
-    // NOTE:change the read lock to the write lock due to the endless-loop issue [ALLUXIO-3089]
-    try (LockResource r = new LockResource(mMetadataWriteLock)) {
-      plan = mEvictor.freeSpaceWithView(availableBytes, location, getUpdatedView(), mode);
-      // Absent plan means failed to evict enough space.
-      if (plan == null) {
-        throw new WorkerOutOfSpaceException(
-            ExceptionMessage.NO_EVICTION_PLAN_TO_FREE_SPACE, availableBytes, location.tierAlias());
+  private void freeSpaceInternal(long sessionId, long minContigiousBytes, long minAvailableBytes,
+      BlockStoreLocation location) throws WorkerOutOfSpaceException, IOException {
+    // TODO(ggezer): Too much memory pressure when pinned-inodes list is large.
+    BlockMetadataEvictorView evictorView = getUpdatedView();
+    LOG.debug(
+        "freeSpaceInternal - locAvailableBytes: {}, minContigiousBytes: {}, minAvailableBytes: {}",
+        evictorView.getAvailableBytes(location), minContigiousBytes, minAvailableBytes);
+    boolean contigiousSpaceFound = false;
+    boolean availableBytesFound = false;
```

Comment:
Do we care about `availableBytesFound`? It seems like the only satisfactory condition is `contigiousSpaceFound`